rxn4chemistry/response_handler.py

```diff
@@ -48,10 +48,14 @@ def handle(self) -> Any:
         successful_status = self.response.status_code == self.success_status_code
         error_in_response_dict = self._response_dict_has_error_status()
 
-        # Success: must be the adequate status code, and there must be no
-        # error in the response body. All other cases are considered to be
-        # an error.
-        if successful_status and not error_in_response_dict:
+        # Success: must be the adequate status code, there must be no
+        # error in the response body and the response must not be None.
+        # All other cases are considered to be an error.
+        if (
+            successful_status
+            and not error_in_response_dict
+            and self.response is not None
```

Comment:
Shouldn't this be `self._payload` instead of `self.response`? (also further down in `_print_error_logs`).

Reply:
In principle yes, in case `self.response == None` => `self._payload == None`. I decided to go to response directly because we did not explicitely check for `self.response` equality to `None` and we pass it to `self.on_success`. But if  you believe it's better to use payload, happy to change, I don't have a strong opinion about it.

Reply:
From the error message that we saw:
```
  File "[...]response_handler.py", line 55, in handle
    return self.on_success(self.response)
  File "[...]callbacks.py", line 133, in automatic_retrosynthesis_results_on_success
    for sequence in response_dict["payload"]["sequences"]
TypeError: 'NoneType' object is not subscriptable
```

It is not `self.response` that is `None`; it is either `self.response.json()` or `self.response.json()["payload"]`. I think that `self.response` is never `None` actually.

My message above assumed it would always be `self._payload`, but actually it could also be `self._response_dict`

Reply:
EDIT: actually it seems to be always `self._payload`, as I don't think that the function `.json()` can return `None`.

Reply:
makes sense, pushing soon the changes